urllib3/util/ssl_.py

```diff
@@ -17,6 +18,26 @@
 
 import errno
 import warnings
+import hmac
+
+if hasattr(hmac, 'compare_digest'):
+    _const_compare_digest = hmac.compare_digest
+else:
+    def _const_compare_digest(a, b):
+        """
+        Compare two digests of equal length in constant time.
+
+        Returns True if the digests match, and False otherwise.
+        """
+        if len(a) != len(b):
+            raise ValueError("Mismatched input lengths!")
+
+        result = 0
+        for l, r in six.moves.zip(bytearray(a), bytearray(b)):
```

Comment:
I also find the 1-liner somewhat more difficult to read, and I don't think the 3 lines saved change that much in the overall project.